packages/ingester/src/ingest.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  EntityType,
+  Event,
+  EventType,
+  generateUniqueID,
+  QATaskContent,
+  Task,
+  TaskContent,
+  TaskContentType,
+  TaskIngestEvent,
+  TaskProvenance,
+  TaskSpecType,
+  TaskUpdateDeletedEvent,
+} from "@withorbit/core";
+import { DatabaseEntityQuery, OrbitStore } from "@withorbit/store-shared";
+import { IngestiblePrompt, IngestibleSource } from "./ingestible";
+
+// TODO: handle moves across sources
+// TODO: handle provenance updates
+export async function ingestSources(
+  sources: IngestibleSource[],
+  store: OrbitStore,
+) {
+  const events: Event[] = [];
+
+  // TODO: create a new query or extend the entity query such that
+  // we can filter on only sources that would have used this ingester
+  const query: DatabaseEntityQuery<Task> = {
+    entityType: EntityType.Task,
+  };
+  const entities = await store.database.listEntities(query);
+  const groupedEntities = groupEntitiesByIdentifiers(entities);
+  const timeMillis = Date.now();
+
+  for (const source of sources) {
+    if (groupedEntities[source.identifier]) {
+      // determine which prompts are new and which have been deleted
+      const existingPrompts = [...groupedEntities[source.identifier]];
+
+      for (const prompt of source.prompts) {
+        // NOTE: this `findIndex` is a bit expensive since it scans of each iteration.
+        // If this becomes a bottleneck we should precompute a hash for each task prompt.
+        const existingPromptIndex = existingPrompts.findIndex(
```

Comment:
This suggestion complicates your design here, but I figure it's worth discussing:

In the present design, ingestion sources surface a bunch of prompt content (i.e. front/back strings). The shared ingestion machinery implements a notion of "identity" on top of this by matching that verbatim content with existing prompt content.

This is expensive, but it might also be more limiting than it needs to be. I implemented that policy for the Markdown importer because that format has no way of tracking identity over time. But Biome could very reasonably make a little ID for each prompt you create in its UI! We could be using those IDs to maintain a prompt's identity even when its text is modified.

Concretely, an alternative would be to have `IngestionPrompt` include an arbitrary identifier, provided by the source. For Markdown prompts, that'd just be a hash of the content; for other sources it might be some stable identifier. We'd store that in a metadata key on the `Task`. Then we could replace this double-loop with set arithmetic.

(Feel free to YAGNI me of course)